docs/user_guide/hopsworks/jupyter.rst

```diff
@@ -0,0 +1,227 @@
+Jupyter Notebooks
+=======================
+
+This tutorial will go over the basics of using Jupyter notebooks on Hopsworks.
+
+Jupyter Notebooks Basics
+-------------------
+
+**Open the Jupyter Service**
+
+Jupyter is provided as a micro-service on Hopsworks and can be found in the main UI inside a project.
+
+.. _jupyter1.png: ../../_images/jupyter1.png
+.. figure:: ../../imgs/jupyter1.png
+    :alt: Open the Jupyter service on Hopsworks
+    :target: `jupyter1.png`_
+    :align: center
+    :figclass: align-center
+
+    Open the Jupyter service on Hopsworks
+
+**Start a Jupyter notebook server**
+
+When you start a Jupyter notebook server you have the possibility to specify Spark properties of the notebooks you will create on the server. Hopsworks provides *Machine learning as a service* with tensorflow and spark, supporting distirbuted training, parallel experiments, hyperparameter tuning, and model serving (HopsML). If you are doing machine learning on hops you probably want to select the notebook servers "Experiment", "Parallel Experiment" or "Distributed training" as shown in the figure below. For more information about *HopsML*, see the hops-tensorflow_ documentation. For general purpose notebooks, select the type "Spark (Static)" or "Spark (Dynamic)".
+
+.. _jupyter2.png: ../../_images/jupyter2.png
+.. figure:: ../../imgs/jupyter2.png
+    :alt: Start a Jupyter notebook server
+    :target: `jupyter2.png`_
+    :align: center
+    :figclass: align-center
+
+    Start a Jupyter notebook server
+
+**Jupyter + Spark on Hopsworks**
+
+As a user, you will just interact with the Jupyter notebooks, but if you are interested about the technologies behind it, it is explained below.
+
+When using Jupyter on Hopsworks, a library called `sparkmagic`_ is used to interact with the Hops (hadoop) cluster. When you create a Jupyter notebook on Hopsworks, you first select a **kernel**. A kernel is simply a program that executes the code that you have in the Jupyter cells, you can think of it as a REPL-backend to your jupyter notebook that acts as a frontend.
+
+Sparkmagic works with a remote REST server for Spark, called `livy`_, running inside the Hops cluster. Livy is an interface that Jupyter-on-Hopsworks use to interact with the Hops cluster. When you run Jupyter cells using the pyspark kernel, the kernel will automatically send commands to livy in the background for executing the commands on the cluster. Thus, the work that happens in the background when you run a Jupyter cell is as follows:
+
+- The code in the cell will first go to the kernel.
+- Next, the kernel kernel sends the code as a HTTP REST request to livy.
+- When receiving the REST request, livy executes the code on the Spark driver in the cluster.
+- If the code is regular python/scala/R code, it will be run inside a python/scala/R interpreter on the Spark driver.
```

Comment:
@Limmen `will be executed` or `will run`

Reply:
Fixed